File: weka/kmeans.java
import weka.clusterers.SimpleKMeans;
import weka.core.*;
import weka.core.converters.ConverterUtils.DataSource;

public class kmeans {
	public static void main(String args[]) {

		try {
			// Load the Data.
			DataSource source = new DataSource("../data/two_cluster.arff");
			Instances data = source.getDataSet();

			// Perform kmeans with 2 clusters.
			SimpleKMeans kmeans = new SimpleKMeans();
			kmeans.setPreserveInstancesOrder(true);
			kmeans.setNumClusters(2);
			kmeans.buildClusterer(data);

			// Show cluster association.
			int[] assignments = kmeans.getAssignments();
			for (int cluster : assignments)
				System.out.println(cluster);

			// Show cluster centers.
			Instances centers = kmeans.getClusterCentroids();
			System.out.println(centers);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: mlpy/kMeans.py
import numpy as np
import os
from mlpy import Kmeans

# Load the data.
f = open(os.path.dirname(__file__) + '../data/two_cluster.data')
data = np.fromfile(f, dtype=np.float64, sep=' ')
data = data.reshape(-1, 2)
f.close()

# Perform kmean with 2 clusters.
kmeans = Kmeans(2)
association = kmeans.compute(data)

# Show cluster association.
print np.matrix(association).T

# Show cluster centers.
print kmeans.means